### PR TITLE
compact code

### PR DESCRIPTION
**Pull request by eumiro**
Refactor to remove unnecessary code pieces, especially in dicts, list/set comprehensions, and conditions.

### CODE REVIEW

File: osmnx/utils_graph.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ def remove_isolated_nodes(G):
     G = G.copy()
 
     # get the set of all isolated nodes, then remove them
-    isolated_nodes = {node for node, degree in G.degree() if degree < 1}
+    isolated_nodes = {node for node, degree in G.degree() if not degree}
```
Comment (gboeing) [inline osmnx/utils_graph.py:307]: Please undo this change. When handling integers, implicit false may involve more risk than benefit and makes the code less readable.

**Review by gboeing — CHANGES_REQUESTED** [review]
@eumiro thanks for offering this contribution to OSMnx. I have commented on a few requested changes.

File: osmnx/utils_graph.py (hunk 1 above)
Comment (eumiro) [reply osmnx/utils_graph.py]: Thank you, done.

File: osmnx/simplification.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def _is_endpoint(G, node, strict=True):
         return True
 
     # rule 2
-    elif G.out_degree(node) == 0 or G.in_degree(node) == 0:
+    elif not G.out_degree(node) or not G.in_degree(node):
```
[the remark this answers is not in the document]
Comment (eumiro) [reply osmnx/simplification.py]: Thank you, done.

File: osmnx/geometries.py [hunk 3]
```diff
@@ -389,7 +389,7 @@ def _create_gdf(response_jsons, polygon, tags):
                     # If the node has tags and the tags are not empty parse it
                     # to a Point. Empty check is necessary for JSONs created
                     # from XML where nodes without tags are assigned tags={}
-                    if "tags" in element and len(element["tags"]) > 0:
+                    if element.get("tags"):
```
[the remark this answers is not in the document]
Comment (eumiro) [reply osmnx/geometries.py]: Thank you, done.

**Review by gboeing — APPROVED** [review]
